src/algebra/big_operators/basic.lean

```diff
@@ -1332,6 +1332,20 @@ begin
   rwa eq_of_mem_of_not_mem_erase hx hnx
 end
 
+@[to_additive]
+lemma ite_exists_eq_prod {f : α → Prop} [Π i, decidable (f i)]
+  (hf : (s : set α).pairwise_disjoint f) (a : β) :
+  ite (∃ i ∈ s, f i) a 1 = ∏ i in s, ite (f i) a 1 :=
```

Comment:
Would this make more sense to rewrite as
```suggestion
  ite (∃ i ∈ s, f i) a 1 = ∏ i in s.filter f, a :=
```

Reply:
I think I want ite as a sum of ite. The whole point of this lemma, as I understand it, is to prove
```
lemma ruesDiffSumOfRuesDiff (n k:ℕ) (h:0<n*k) (m:ℤ) (z:ℂ) : ruesDiff n m z = ∑ k₀ in range k, ruesDiff (n*k) (n*k₀+m) z:=
```
when the following lemma is available:
```
lemma ruesDiffTsumForm (n:ℕ) (m:ℤ) (z:ℂ) : ruesDiff n m z = tsum (λ (k:ℕ), if ((k:ℤ)+m)%n=0 then z ^ k / k.factorial else 0) :=
```
The comparing the coefficients shows an ite that needs to be shown equal to a sum of ite.
Further details at https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/A.20function.20that.20is.20a.20sum.20of.20said.20function

Reply:
Would this make more sense if the two sides are reversed? This is how you describe it in the PR description as well.

Reply:
@YaelDillies, could you please say what you want to do with this conversation?

Reply:
I reversed the equality as Andrew suggested and kept the `∏ i in s, ite (f i) a 1` spelling as Eric suggested that otherwise it could be simplified using [`finset.prod_filter`](https://leanprover-community.github.io/mathlib_docs/algebra/big_operators/basic.html#finset.prod_filter).